check/check.go

```diff
@@ -78,6 +78,7 @@ type Check struct {
 	ActualValue    string `json:"actual_value"`
 	Scored         bool   `json:"scored"`
 	ExpectedResult string `json:"expected_result"`
+	WarnReason     string `json:"warn_reason,omitempty"`
```

Comment:
I'm tempted to go with the suggestion that this should be called `Reason` with the json version `reason` rather than `warn_reason`. That way if we ever want to give any other supporting information for a failure (say) we could potentially use this field.